packages/babel-parser/src/parser/util.js

```diff
@@ -175,26 +176,28 @@ export default class UtilParser extends Tokenizer {
     /* eslint-enable @babel/development-internal/dry-error-messages */
   }
 
-  expectPlugin(name: string, pos?: ?number): true {
-    if (!this.hasPlugin(name)) {
+  expectPlugin(pluginConfig: PluginConfig, pos?: ?number): true {
+    if (!this.hasPlugin(pluginConfig)) {
       throw this.raiseWithData(
         pos != null ? pos : this.state.start,
-        { missingPlugin: [name] },
-        `This experimental syntax requires enabling the parser plugin: '${name}'`,
+        { missingPlugin: [pluginConfig] },
+        `This experimental syntax requires enabling the parser plugin: ${JSON.stringify(
+          pluginConfig,
+        )}.`,
       );
     }
 
     return true;
   }
 
-  expectOnePlugin(names: Array<string>, pos?: ?number): void {
-    if (!names.some(n => this.hasPlugin(n))) {
+  expectOnePlugin(pluginConfigs: Array<PluginConfig>, pos?: ?number): void {
+    if (!pluginConfigs.some(c => this.hasPlugin(c))) {
       throw this.raiseWithData(
         pos != null ? pos : this.state.start,
-        { missingPlugin: names },
-        `This experimental syntax requires enabling one of the following parser plugin(s): '${names.join(
-          ", ",
-        )}'`,
+        { missingPlugin: pluginConfigs },
```

Comment:
`missingConfig` must always be a string, because that's part of the public API and `@babel/core` relies on it to suggest which `@babel/` plugin should be enabled.. I would prefer to only extend `.hasPlugin`, and leave `.expectPlugin` as-is.

Reply:
If `missingPlugin` with `raiseWithData` is the problem, then we can fix this line, but does that mean `expectPlugin` can’t be changed? `@babel/core` relies on `missingPlugin` in `raiseWithData`, but does it rely directly on `expectPlugin`?

Reply:
I have changed the reviewed code so that `raiseWithData`’s `missingPlugin` only accepts arrays of strings again, so that `@babel/core` doesn’t break. I kept the type extension to `expectPlugin`. Is this okay now?

(We should also look into adding a unit test, in a separate pull request, to make sure that nobody else breaks `raiseWithData`’s `missingPlugin` in the future.)